apps/prairielearn/src/ee/pages/institutionAdminLti13/institutionAdminLti13.ts

```diff
@@ -0,0 +1,192 @@
+import { Router } from 'express';
+import asyncHandler = require('express-async-handler');
+import jose = require('node-jose');
+import { z } from 'zod';
+import error = require('@prairielearn/error');
+import { loadSqlEquiv, queryAsync, queryRows } from '@prairielearn/postgres';
+import { flash } from '@prairielearn/flash';
+
+import { getInstitution } from '../../lib/institution';
+import { InstitutionAdminLti13 } from './institutionAdminLti13.html';
+import { Lti13InstanceSchema } from '../../../lib/db-types';
+import { getCanonicalHost } from '../../../lib/url';
+
+const sql = loadSqlEquiv(__filename);
+const router = Router({ mergeParams: true });
+
+// Middlewear to check for feature and access
+router.use(
+  asyncHandler(async (req, res, next) => {
+    if (!res.locals.lti13_enabled) {
+      throw error.make(403, 'Access denied (feature not available)');
+    }
+    next();
+  }),
+);
+
+router.get(
+  '/:lti13_instance_id?',
+  asyncHandler(async (req, res) => {
+    const institution = await getInstitution(req.params.institution_id);
+    const lti13Instances = await queryRows(
+      sql.select_instances,
+      {
+        institution_id: req.params.institution_id,
+      },
+      Lti13InstanceSchema,
+    );
+
+    const platform_defaults = await queryRows(
+      sql.select_defaults,
+      {},
+      z.object({
+        platform: z.string(),
+        issuer_params: z.any(),
+      }),
+    );
+
+    // Handle the / (no id passed case)
+    if (typeof req.params.lti13_instance_id === 'undefined' && lti13Instances.length > 0) {
+      return res.redirect(`lti13/${lti13Instances[0].id}`);
+    }
+
+    const paramInstance = lti13Instances.find(({ id }) => id === req.params.lti13_instance_id);
+
+    res.send(
+      InstitutionAdminLti13({
+        institution,
+        lti13Instances,
+        instance: paramInstance ? paramInstance : null,
+        resLocals: res.locals,
+        platform_defaults,
+      }),
+    );
+  }),
+);
+
+router.post(
+  '/:lti13_instance_id?',
```

Comment:
Conventionally, we prefix route params with `unsafe_` to indicate that they aren't trusted (that is, someone could edit a URL to include an LTI instance ID that's unrelated to the current institution.